src/libPMacc/include/particles/tasks/TaskParticlesReceive.hpp

```diff
@@ -49,15 +57,25 @@ namespace PMacc
         {
             state = Init;
             EventTask serialEvent = __getTransactionEvent();
+            HandleExchanged handleExchanged;
+            HandleNotExchanged handleNotExchanged;
 
             for (int i = 1; i < Exchanges; ++i)
             {
-                if (parBase.getParticlesBuffer().hasReceiveExchange(i))
-                {
+                /* Start new transaction */
+                if(canRunInParallel)
```

Comment:
The currently used policies set this to false. So the "known version" is used here. If the policies only read the frames, __startTransaction would be possible here. But currenlty it is not used, but only a preparation for future implementations.